packages/eslint-plugin/docs/rules/no-unsafe-type-assertion.mdx

```diff
@@ -0,0 +1,59 @@
+---
+description: 'Disallow type assertions that widen a type.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unsafe-type-assertion** for documentation.
+
+This rule forbids using type assertions to narrow a type, as it can lead to unsafe assumptions that bypass TypeScript’s type-checking.
+
+Instead, it's better to rely on type guards to ensure type safety and avoid potential runtime errors caused by unsafe type assertions.
+
+## Examples
+
+<Tabs>
+<TabItem value="❌ Incorrect">
+
+```ts
+function f() {
+  return Math.random() < 0.5 ? 42 : 'oops';
+}
+
+const z = f() as number;
+
+const items = [1, '2', 3, '4'];
+
+const number = items[0] as number;
+```
+
+</TabItem>
+<TabItem value="✅ Correct">
+
+```ts
+function f() {
+  return Math.random() < 0.5 ? 42 : 'oops';
+}
+
+const z = f() as number | string | boolean;
+
+const items = [1, '2', 3, '4'];
+
+const number = items[0] as number | string | undefined;
+```
+
+</TabItem>
+</Tabs>
+
+## When Not To Use It
```

Comment:
could be worth noting that this will cause lots of reports in test code if you stub objects out, too

Reply:
Oh, this can definitely create some noise. I added a couple of sentences to explain this.